pkg/syndesis/action/install.go

```diff
@@ -24,29 +31,45 @@ func (a *Install) Execute(syndesis *v1alpha1.Syndesis) error {
 
 	logrus.Info("Installing Syndesis resource ", syndesis.Name)
 
-	sa := newSyndesisServiceAccount()
-	setNamespaceAndOwnerReference(sa, syndesis)
-	// We don't replace the service account if already present, to let Kubernetes generate its tokens
-	err := sdk.Create(sa)
-	if err != nil && !k8serrors.IsAlreadyExists(err) {
+	token, err := installServiceAccount(syndesis)
+	if err != nil {
 		return err
 	}
 
-	token, err := serviceaccount.GetServiceAccountToken(sa.Name, syndesis.Namespace)
+	list, err := syndesistemplate.GetInstallResourcesAsRuntimeObjects(syndesis, syndesistemplate.InstallParams{
+		OAuthClientSecret: token,
+	})
 	if err != nil {
 		return err
 	}
 
-	list, err := syndesistemplate.GetInstallResources(syndesis, syndesistemplate.InstallParams{
-		OAuthClientSecret: token,
-	})
+	syndesisRoute, err := installSyndesisRoute(syndesis, list)
+	if err != nil {
+		return err
+	}
 
-	for _, obj := range list {
-		res, err := util.LoadKubernetesResource(obj.Raw)
+	if syndesis.Spec.RouteHostName == "" {
+		syndesis.Spec.RouteHostName = syndesisRoute.Spec.Host
+
+		// Hack to remove the auto-generated annotation
+		// In Openshift 3.9, the route gets low priority for being displayed as main route for the app if the openshift.io/host.generated=true annotation is present
+		err = removeAutoGeneratedAnnotation(syndesisRoute)
 		if err != nil {
 			return err
 		}
 
+		// Recreate the list of resources to inject the route hostname
+		list, err = syndesistemplate.GetInstallResourcesAsRuntimeObjects(syndesis, syndesistemplate.InstallParams{
```

Comment:
I see, the host is required here. So maybe we should do the `target` copy a bit earlier ?